server.go

```diff
@@ -95,6 +95,10 @@ const (
 	// value used or a particular peer will be chosen between 0s and this
 	// value.
 	maxInitReconnectDelay = 30
+
+	// multiAddrConnectionStagger is the number of seconds to wait between
```

Comment:
do you mean the using the `maxInitReconnectDelay` as the stagger amount as an alternative or do you mean using the existing `delayInitialReconnect` function as the stagger method?

The reason i didn't use those is that those were made specifically to stagger connections requests to multiple peers where as this is to stagger connection requests to the same peer.
-  So I thought that 30 seconds was a bit long to wait between attempting the different addresses of the same peer (maybe 10 seconds is also too long? evidenced by the fact that testing this worked even when i accidentally only used 10ns stagger 🙈 ) like if one addr fails, we dont want to wait too long before trying the next one. 
- the `delayInitalReconnect` function cant be used for this imo because since this issue is about staggering multiple connection reqs to the same peer meaning that if one succeeds, we need a way to cancel all the others immediately